internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -298,6 +300,73 @@ func (f *fleetGateway) doExecute(ctx context.Context, bo backoff.Backoff) (*flee
 	return nil, ctx.Err()
 }
 
+func (f *fleetGateway) convertToCheckinComponents(components []runtime.ComponentComponentState) []fleetapi.CheckinComponent {
+	if components == nil {
+		return nil
+	}
+	stateString := func(s eaclient.UnitState) string {
+		switch s {
+		case eaclient.UnitStateStarting:
+			return "starting"
+		case eaclient.UnitStateConfiguring:
+			return "configuring"
+		case eaclient.UnitStateHealthy:
+			return "healthy"
+		case eaclient.UnitStateDegraded:
+			return "degraded"
+		case eaclient.UnitStateFailed:
+			return "failed"
+		case eaclient.UnitStateStopping:
+			return "stopping"
+		case eaclient.UnitStateStopped:
+			return "stopped"
+		}
+		return ""
+	}
```

Comment:
@blakerouse @aleksmaus I believe we have a similar lock elsewhere (inspect or diagnostic) maybe we should have an helper function that takes the UnitState and return a human readable representation?

Reply:
I think for inspect and diagnostics the Stringer interface that I added to elastic-agent-client works in that situation. And really it produces the same result as here, but this will be an API interface with Fleet Server so coding it here seems safer.

Reply:
OK, This is a valid point.